File: single-language-analysis/dataset-files/python/test_smell_099.py
# Original URL: https://github.com/ogrisel/scikit-learn/blob/b20998d4ec606558ae86cba7ff5a84aea094c397/sklearn/cluster/tests/test_k_means.py#L381-L405

@pytest.mark.parametrize("algorithm", ["lloyd", "elkan"])
@pytest.mark.parametrize("tol", [1e-2, 0])
def test_smell_099(algorithm, tol, capsys):
    # Check verbose mode of KMeans for better coverage.
    X = np.random.RandomState(0).normal(size=(5000, 10))

    KMeans(
        algorithm=algorithm,
        n_clusters=n_clusters,
        random_state=42,
        init="random",
        n_init=1,
        tol=tol,
        verbose=1,
    ).fit(X)

    captured = capsys.readouterr()

    assert re.search(r"Initialization complete", captured.out)
    assert re.search(r"Iteration [0-9]+, inertia", captured.out)

    if tol == 0:
        assert re.search(r"strict convergence", captured.out)
    else:
        assert re.search(r"center shift .* within tolerance", captured.out)

File: single-language-analysis/dataset-files/java/TestSmell099.java
import org.junit.Test;
import static org.junit.Assert.*;
import java.util.regex.Pattern;

public class TestSmell099 {

    @Test
    public void testKMeansVerbose() {
        String[] algorithms = {"lloyd", "elkan"};
        double[] tols = {1e-2, 0};

        for (String algorithm : algorithms) {
            for (double tol : tols) {
                double[][] X = new double[5000][10];
                // Populate X with random values (omitted for brevity)

                KMeans kMeans = new KMeans(
                    algorithm,
                    nClusters,
                    42,
                    "random",
                    1,
                    tol,
                    1
                );
                kMeans.fit(X);

                String captured = ""; // Capture output (omitted for brevity)

                assertTrue(Pattern.compile("Initialization complete").matcher(captured).find());
                assertTrue(Pattern.compile("Iteration [0-9]+, inertia").matcher(captured).find());

                if (tol == 0) {
                    assertTrue(Pattern.compile("strict convergence").matcher(captured).find());
                } else {
                    assertTrue(Pattern.compile("center shift .* within tolerance").matcher(captured).find());
                }
            }
        }
    }
}